guava/src/com/google/common/base/Converter.java

```diff
@@ -165,12 +167,23 @@ protected Converter() {
    * @return the converted value; is null <i>if and only if</i> {@code a} is null
    */
   @CanIgnoreReturnValue
-  public final @Nullable B convert(@Nullable A a) {
+  public final @PolyNull B convert(@PolyNull A a) {
     return correctedDoForward(a);
   }
 
-  @Nullable
-  B correctedDoForward(@Nullable A a) {
+  @SuppressWarnings({
+    "return.type.incompatible",
+    /*
+     * If handleNullAutomatically is true then correctedDoForward returns null only if 'a' is null
+     * else it returns a non-null value. Known behaviour of @PolyNull with ternary operator, please
+     * refer to link: https://github.com/typetools/checker-framework/issues/1170
+     */
+    "argument.type.incompatible"
+    /*
+     * If handleNullAutomatically is false, then argument can be assumed to be non-null
```

Comment:
Why can this be assumed?  Please explain the reasoning.

Reply:
Here is the link to comments on `handleNullAutomatically` in class [com.google.common.base.Converter](https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Converter.java), see, https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Converter.java#L128-L131. If documentation needs refinement, please let me know so that I can pull request the changes for the same.